### ISSUE TITLE
Cache image reponses are not disposed

### ISSUE THREAD

**Issue by deanmarcussen** [SixLabors/ImageSharp.Web#162]
### Prerequisites

- [x] I have written a descriptive issue title
- [x] I have verified that I am running the latest version of ImageSharp.Web
- [x] I have verified if the problem exist in both `DEBUG` and `RELEASE` mode
- [x] I have searched [open](https://github.com/SixLabors/ImageSharp.Web/issues) and [closed](https://github.com/SixLabors/ImageSharp.Web/issues?q=is%3Aissue+is%3Aclosed) issues to ensure it has not already been reported

### Description

There is a couple of reported memory related issues, and while I know there is some tracking in ImageSharp itself about releasing retained resources, there looks to be another issue here (easily resolved) regarding stream usage.

Reported here on Orchard Core https://github.com/OrchardCMS/OrchardCore/issues/9060
and likely related to this discussion https://github.com/SixLabors/ImageSharp.Web/discussions/161

Performance testing locally this causes memory usage to grow dramatically - got as high as 15Gb before I stopped it.

Collecting manually doesn't drop it, so I suspect that's why there's reported app service failures.

### Steps to Reproduce

This line doesn't dispose of the cache file streams
https://github.com/SixLabors/ImageSharp.Web/blob/8f67aae9d314a6fde96aae856027a655d6cbd2fa/src/ImageSharp.Web/Middleware/ImageSharpMiddleware.cs#L476

adding a dispose
```
                    using (var stream = await cacheResolver.OpenReadAsync())
                    {
                        await imageContext.SendAsync(stream, metadata);
                    }
```

and everything gets collected when the GC runs, and memory usage is as one would expect.

I've seen some (difficult to reproduce) issues with file locks on the cache images, but I am hoping this might explain those as well.

Off for the covid jab in a minute, so pr to come

### System Configuration

Just locally.

**Comment by deanmarcussen** [comment]
Note: that sample code doesn't match well as I'm working of this pr at the moment for testing https://github.com/SixLabors/ImageSharp.Web/pull/160

**Comment by JimBobSquarePants** [comment]
Great catch! I have time so will PR right now.

**Comment by JimBobSquarePants** [comment]
Fixed by #160
